Corrupted pachyderm config segfaults pachctl

<!-- If this is a bug, fill this section out -->

**What happened?**:
```
WARN[0000] error loading user config from ~/.pachderm/config: unexpected end of JSON input 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xab6af0]

goroutine 1 [running]:
github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client.NewOnUserMachine(0xc4201f0201, 0x16175c9, 0x4, 0x0, 0x0, 0x0, 0x0, 0xc4204b3b50, 0x4c5df4)
        /home/jdoliner/go/src/github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/client.go:306 +0x200
github.com/pachyderm/pachyderm/src/server/pfs/cmds.Cmds.func6(0x2185340, 0x0, 0x0, 0x412358, 0x10)
        /home/jdoliner/go/src/github.com/pachyderm/pachyderm/src/server/pfs/cmds/cmds.go:136 +0x7c
github.com/pachyderm/pachyderm/src/server/pkg/cmdutil.RunFixedArgs.func1(0xc42042d680, 0x2185340, 0x0, 0x0)
        /home/jdoliner/go/src/github.com/pachyderm/pachyderm/src/server/pkg/cmdutil/cobra.go:20 +0x134
github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra.(*Command).execute(0xc42042d680, 0x2185340, 0x0, 0x0, 0xc42042d680, 0x2185340)
        /home/jdoliner/go/src/github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra/command.go:572 +0x234
github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420240900, 0xc4201a0fe5, 0x12e6380, 0xc4201a068d)
        /home/jdoliner/go/src/github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra/command.go:658 +0x2d4
github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420240900, 0x0, 0x0)
        /home/jdoliner/go/src/github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra/command.go:617 +0x2b
main.main.func1(0x15b9a40, 0xc4203be480)
        /home/jdoliner/go/src/github.com/pachyderm/pachyderm/src/server/cmd/pachctl/main.go:21 +0x4a
main.main()
        /home/jdoliner/go/src/github.com/pachyderm/pachyderm/src/server/cmd/pachctl/main.go:22 +0xbe
```

**What you expected to happen?**:

Pachctl should regenerate the config for me

**How to reproduce it (as minimally and precisely as possible)?**:

Put malformed json in ~/.pachyderm.config

--- a/src/client/client.go
+++ b/src/client/client.go
@@ -354,10 +354,10 @@ func NewOnUserMachine(reportMetrics bool, prefix string, options ...Option) (*AP
 
 	// Add metrics info & authentication token
 	client.metricsPrefix = prefix
-	if cfg.UserID != "" && reportMetrics {
+	if cfg != nil && cfg.UserID != "" && reportMetrics {
 		client.metricsUserID = cfg.UserID
 	}
-	if cfg.V1 != nil && cfg.V1.SessionToken != "" {
+	if cfg != nil && cfg.V1 != nil && cfg.V1.SessionToken != "" {
 		client.authenticationToken = cfg.V1.SessionToken
 	}
 	return client, nil

